converters/fromarray.ts

```diff
@@ -0,0 +1,59 @@
+import {  IEvent, IEventArray, ILayoutState } from "../clarity";
+import { Action, ObjectType } from "../clarity";
+import schemas from "./schema";
+
+export default function(eventArray: IEventArray): IEvent {
+  let id          = eventArray[0];
+  let type        = eventArray[1];
+  let time        = eventArray[2];
+  let stateArray  = eventArray[3];
+  let schema      = eventArray[4];
```

Comment:
You could use a struct of array rather than array of structs for encdoing. That is have something like:

{
id: [1, 5, ...]
time: [10, 20, 30]
type: ['a', 'b', ]
schema: ['schema1', ....]
state: [....]
}

Probably compresses@  better and consumes less memory.